lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -34,11 +34,62 @@ static int getTensorRank(Value tensor) {
   return tensorRank;
 }
 
+// Helper function to compute the return type of the reduction function along
+// multiple dimensions. `dimList` specifies a list of dimensions to reduce and
+// `keepDim` preserves the rank of the input tensor.
+static LogicalResult
```

Comment:
nit: You can use the return type `FailureOr<BaseTensorType>` to return a `rewriter.notifyMatchFailure` or a `BaseTensorType`, so that you don't have to pass the return value as an argument to the function. You can then check if the returned value failed by doing `if (failed(LogicalResult(returnedValue)))`